### PR TITLE
feat: add 'Generate summary' integration: frontend ✨

### LINKED ISSUE #13430: AI generated unread messages summaries
When we have a lot of unread messages in a conversation, it would be nice to have a short summary of them.

## Expectation

A button that allows to generate summaries would appear next to the unread messages divider. Clicking it would start the action

Image

An "info" badge would appear in the new message form, indicating that the task has started

Image

And finally, when the summary is generated, it could look like this.

Image
 
>[!WARNING]
> The summary box **MUST** contain a disclaimer that the summary is AI generated and can contain mistakes

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Ref #13430
* refactor unread marker styles

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
image | imageimage

image
image

### 🏁 Checklist

- [x] 🌏 Tested with different browsers / clients:
  - [x] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [x] Talk Desktop
  - [ ] Not risky to browser differences / client
- [x] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/NewMessage/NewMessageChatSummary.vue [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+			const summary = response.data.ocs.data.task.output?.output ?? t('spreed', 'Empty summary response')
+			chatExtrasStore.storeChatSummary(token, task.fromMessageId, summary)
+			clearInterval(getTaskInterval)
+			getTaskInterval = undefined
+			checkScheduledTasks(token)
+			break
+		}
+		case TASK_PROCESSING.STATUS.FAILED:
+		case TASK_PROCESSING.STATUS.UNKNOWN:
+		case TASK_PROCESSING.STATUS.CANCELLED: {
```
[the remark this answers is not in the document]
Comment (DorraJaouad) [reply src/components/NewMessage/NewMessageChatSummary.vue:163]: At least show a warning on top of the final output "This summary may not be complete"

**Review by DorraJaouad — APPROVED** [review]

File: src/components/MessagesList/MessagesGroup/Message/Message.vue [hunk 2]
```diff
@@ -222,9 +235,20 @@ export default {
...
+
+			if (this.message.id === this.visualLastLastReadMessageId) {
+				return !this.isCollapsedSystemMessage || this.message.id !== this.lastCollapsedMessageId
+			}
+
+			return this.isCombinedSystemMessage && this.lastCollapsedMessageId === this.visualLastLastReadMessageId
+		},
+
+		shouldShowSummaryOption() {
+			if (!canSummarizeChat || !this.isLastReadMessage
```
Comment (DorraJaouad) [inline src/components/MessagesList/MessagesGroup/Message/Message.vue:247]:
```suggestion
			if (!canSummarizeChat
```

It's already in the unread marker div

File: src/components/NewMessage/NewMessageChatSummary.vue [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+onMounted(() => {
+	setIsTextMoreThanOneLine()
+})
+
+onBeforeUnmount(() => {
+	Object.values(cancelGetTask).forEach(cancelFn => cancelFn())
+})
+
+watch(token, (newValue, oldValue) => {
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/NewMessage/NewMessageChatSummary.vue:111]: Will reset to '', then checkScheduledTasks should set correct value

File: src/components/NewMessage/NewMessageChatSummary.vue [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+ * @param task task object
+ */
+async function getTask(token: string, request: TaskProcessingCancelableRequest['request'], task: ChatTask) {
+	try {
+		const response = await request(task.taskId)
+		const status = response.data.ocs.data.task.status
+		switch (status) {
+		case TASK_PROCESSING.STATUS.SUCCESSFUL: {
+			// Task is completed, proceed to the next task
+			const summary = response.data.ocs.data.task.output?.output ?? t('spreed', 'Empty summary response')
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/NewMessage/NewMessageChatSummary.vue:166]: Follow-up

File: src/components/NewMessage/NewMessageChatSummary.vue [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+			clearInterval(getTaskInterval)
+			getTaskInterval = undefined
+			checkScheduledTasks(token)
+			break
+		}
+		case TASK_PROCESSING.STATUS.FAILED:
+		case TASK_PROCESSING.STATUS.UNKNOWN:
+		case TASK_PROCESSING.STATUS.CANCELLED: {
+			// Task is likely failed, proceed to the next task
+			chatExtrasStore.storeChatSummary(token, task.fromMessageId, t('spreed', 'Error occurred during a summary generation'))
```
Comment (DorraJaouad) [inline src/components/NewMessage/NewMessageChatSummary.vue:180]:
```suggestion
			chatExtrasStore.storeChatSummary(token, task.fromMessageId, '')
```

File: src/stores/chatExtras.js [hunk 6]
```diff
@@ -61,6 +64,19 @@ export const useChatExtrasStore = defineStore('chatExtras', {
 		getNextEvent: (state) => (token) => {
 			return state.upcomingEvents[token]?.[0]
 		},
+
+		getChatSummaryTaskQueue: (state) => (token) => {
+			return Object.values(Object(state.chatSummary[token]))
+		},
```
Comment (DorraJaouad) [inline src/stores/chatExtras.js:70]:
```suggestion
		},
		
		hasTaskFailed: (state) => (token) => {
			return Object.values(Object(state.chatSummary[token])).any(task => task.status === TASK_PROCESSING.FAILED)
		},
```

Use this to show the hint that sthg went wrong in summarizing. If you think that there are many hints (along with AI disclaimer), then don't show the summary, stop any tasks after the failed ones and show the failed summary text.

Comment (Antreesy) [reply src/stores/chatExtras.js:70]: Follow-up

**Pull request merged**